java/rocksjni/rocksjni.cc

```diff
@@ -1618,7 +1618,7 @@ jboolean key_may_exist_helper(JNIEnv* env, rocksdb::DB* db,
   delete[] key;
 
   // extract the value
-  if (value_found && !value.empty()) {
+  if (value_found && !value.empty() && jstring_builder != nullptr) {
```

Comment:
I am not sure that this is a good idea, as it deviates from the C++ API which we try to stay close to. In the C++ API passing `null` as the `value` (in this case the equivalent of `jstring_builder`) would cause an invalid pointer de-reference and presumably result in a SIGSEGV.

The C++ function implementation , simply contains `assert(value != nullptr);`. Perhaps instead of this we should just add an `assert value != null;` in the Java `keyMayExist` functions?